include/mbedtls/threading.h

```diff
@@ -39,16 +39,72 @@ extern "C" {
 #define MBEDTLS_ERR_THREADING_BAD_INPUT_DATA              -0x001C  /**< Bad input parameters to function. */
 #define MBEDTLS_ERR_THREADING_MUTEX_ERROR                 -0x001E  /**< Locking / unlocking / free failed with error code. */
 
```

Comment:
I think the preprocessor check below is missing from this file:
```
#if defined(MBEDTLS_THREADING_C)
...
#endif /* MBEDTLS_THREADING_C */
```
Not doing this is what caused link errors instead of compilation errors for some users. Please refer to issue https://github.com/ARMmbed/mbedtls/issues/660